File: string-handling/string-characteristics/python/change_string.py
# Isaac Computer Science
# Usage licensed under the Open Government Licence v3.0

def change_string(my_string):
    """Changes all lowercase a to lowercase b"""
    new_string = ""  # New empty string
    for char in my_string:
        if char == "a":
            new_string = new_string + "b"
        else:
            new_string = new_string + char
    return new_string


def test():
    """Test data"""
    my_string = "An aardvark is an animal"
    new_string = change_string(my_string)
    print(new_string)       

   
# This code will run if this file is executed directly
# (i.e. not called by another program)
if __name__ == '__main__':
    test()

File: string-handling/string-mutability/c-sharp/change_string.cs
/*
Isaac Computer Science
Usage licensed under the Open Government Licence v3.0

Note: This file is designed to be copied out and compiled on your machine.
In order for it to compile properly you need to ensure that the project name is the same as the "namespace" in this file.
To run this file you need to:
1. Copy the contents
2. Paste them into the C# IDE of your choice (Visual Studio, for example)
3. Change the namespace to match your project (if necessary)
4. Compile the program
5. Run the program
*/

using System;

namespace IsaacCodeSamples
{

    class StringMutability
    {

        // The Main method is the entry point for all C# programs
        public static void Main() {
            string myString = "An aardvark is an animal";
            string newString = ChangeString(myString);
            Console.WriteLine(newString);
        }


        // Demonstrates how to use ASCII codes to convert letter case
        public static string ChangeString(string myString) {
            string newString = "";  // New empty string
            foreach (var character in myString)
            {
                if (character == 'a') {
                    newString = newString + "b";
                } else {
                    newString = newString + character;
                }
            }
            return newString;
        }


    }
}
